crates/bevy_ecs/src/world/unsafe_world_cell.rs

```diff
@@ -70,75 +72,203 @@ use std::any::TypeId;
 /// }
 /// ```
 #[derive(Copy, Clone)]
-pub struct UnsafeWorldCell<'w>(&'w World);
+pub struct UnsafeWorldCell<'w>(*mut World, PhantomData<(&'w World, &'w UnsafeCell<World>)>);
```

Comment:
We need to actually use the `'w` lifetime inside the struct or else rust gives a compile error, it probably could just be `PhantomData<&'w World>` or `PhantomData<&'w UnsafeCell<World>>` instead of both but i figured it probably doesn't hurt to do both